src/java.base/share/classes/jdk/crac/Core.java

```diff
@@ -173,10 +175,22 @@ private static void checkpointRestore1(long jcmdStream) throws
         }
 
         if (newProperties != null && newProperties.length > 0) {
-            Arrays.stream(newProperties).map(propStr -> propStr.split("=", 2)).forEach(pair -> {
-                AccessController.doPrivileged(
-                    (PrivilegedAction<String>)() ->
-                        System.setProperty(pair[0], pair.length == 2 ? pair[1] : ""));
+            Arrays.stream(newProperties).map(new Function<String, String[]>() {
```

Comment:
A comment reffering to the issue and describing the solution (do not use lambda) would be helpful!